bullet-featherstone/src/FreeGroupFeatures.cc

```diff
@@ -85,37 +85,39 @@ Identity FreeGroupFeatures::FindFreeGroupForModel(
 {
   const auto *model = this->ReferenceInterface<ModelInfo>(_modelID);
 
-  // Reject if the model has fixed base
-  if (model->body->hasFixedBase())
-    return this->GenerateInvalidId();
-
-  // Also reject if the model is a child of a fixed constraint
-  // (detachable joint)
   for (const auto & joint : this->joints)
   {
+    // Also reject if the model is a child of a fixed constraint
+    // (detachable joint)
     if (joint.second->fixedConstraint)
     {
       if (joint.second->fixedConstraint->getMultiBodyB() == model->body.get())
       {
         return this->GenerateInvalidId();
       }
     }
+    // Reject if the model has a world joint
+    if (std::size_t(joint.second->model) == std::size_t(_modelID))
```

Comment:
I think this is good for matching the behavior of DART, but we might want to consider the allowing models with world joints to move. I think it would be more consistent with also allowing static models to move. I won't block on this, but I'd like to hear @scpeters thoughts on it.

Reply:
according to our documentation of `FreeGroup` in [FreeGroup.hh](https://github.com/gazebosim/gz-physics/blob/gz-physics8_8.0.0/include/gz/physics/FreeGroup.hh#L33-L34):

> the FreeGroup class, which represents a group of links that are not connected to the world with any kinematic constraints

I think it could be a nice user feature in gz-sim to be able to move objects that have a joint connecting them to the world, but by the definition of a FreeGroup, I don't think we can support that with this gz-physics API. I think we would need a gz-sim feature that detaches a joint, moves, and then reattaches the joint.

Reply:
ok sounds like a new feature to add  so I'll keep this PR as is.